### PR TITLE
Add a new version CUDA Toolkit 11.0.2 for Arm to cuda package

### PR DESCRIPTION
**Pull request by ikitayama**
Tested on ThunderX2.

### CODE REVIEW

**Comment by ikitayama** [comment]
Could @ax3l take a look?

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ class Cuda(Package):
     # https://www.nvidia.com/en-us/drivers/cuda/mac-driver-archive/ mention
     # Mojave support -- only macOS High Sierra 10.13 is supported.
     conflicts('arch=darwin-mojave-x86_64')
-
+    depends_on('gcc@:9.3.2')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/cuda/package.py]: We handle compiler conflicts in the cuda mixin class (in lib/), please do not add here.

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ class LibjpegTurbo(Package):
     """
     # https://github.com/libjpeg-turbo/libjpeg-turbo/blob/master/BUILDING.md
     homepage = "https://libjpeg-turbo.org/"
-    url = "https://github.com/libjpeg-turbo/libjpeg-turbo/archive/2.0.3.tar.gz"
+    url = "https://github.com/libjpeg-turbo/libjpeg-turbo/archive/2.0.4.tar.gz"
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: This seems unrelated to this PR :)

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 
 _versions = {
     '11.0.2': {
+        'Linux-aarch64': ('3942f368a9afab99ced1df20168f02f5', 'http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux_sbsa.run'),
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/cuda/package.py]: Can you please use the sha256 hash for consistency? :)

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 1 above)
Comment (ikitayama) [reply var/spack/repos/builtin/packages/cuda/package.py]: But without that cuda can get built with gcc@10.1 or higher.

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py (hunk 2 above)
Comment (ikitayama) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: No, I didn't think it was included.

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 3 above)
Comment (ikitayama) [reply var/spack/repos/builtin/packages/cuda/package.py]: Sure I will.

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 1 above)
Comment (ikitayama) [reply var/spack/repos/builtin/packages/cuda/package.py]: I mean, cube can be built presumably whatever the recent GCC but it causes a run time error.

**Comment by ikitayama** [comment]
> Could @ax3l take a look?

Hi @ax3l , I am away from the office until the end of August, as the changes obviously minor would you take the ownership
of this PR is trying to accomplish? Many thanks!

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 1 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/cuda/package.py]: the cuda package does not get "built" in the common sense when it gets installed, it just extracts things. conflicts need to be handled for dependent packages that use the Spack `cuda` package with a selected `%hostCompiler`.

**Comment by ax3l** [comment]
Hi, sorry for the delay, I was on PTO.
I have trouble pushing to your branch, will try to edit on github.

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 3 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/cuda/package.py]:
```suggestion
        'Linux-aarch64': ('23851e30f7c47a1baad92891abde0adbc783de5962c7480b9725198ceacda4a0', 'http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux_sbsa.run'),
```

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 1 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/cuda/package.py]:
```suggestion

```

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py (hunk 2 above)
Comment (ax3l) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]:
```suggestion
    url = "https://github.com/libjpeg-turbo/libjpeg-turbo/archive/2.0.3.tar.gz"
```

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 4]
```diff
@@ -76,7 +77,6 @@ class Cuda(Package):
     # https://www.nvidia.com/en-us/drivers/cuda/mac-driver-archive/ mention
     # Mojave support -- only macOS High Sierra 10.13 is supported.
     conflicts('arch=darwin-mojave-x86_64')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/cuda/package.py]:
```suggestion
    conflicts('arch=darwin-mojave-x86_64')

```

**Review by ax3l — APPROVED** [review]
That should do it, the GCC support is already set in `lib/spack/spack/build_systems/cuda.py`.

**Comment by ikitayama** [comment]
> That should do it, the GCC support is already set in `lib/spack/spack/build_systems/cuda.py`.

I see that that is where the GCC support should be described.

**Comment by sethrj** [comment]
@ax3l I'm guessing you assigned this to me to merge? I'm on the spack team but not authorized for merge privileges. I'd be happy to take the extra responsibility though if someone adds me.

**Review by alalazo — APPROVED** [review]

**Comment by ikitayama** [comment]
I've just `install`ed the package from the develop onto ThunderX2 without an issue!
